Adding in CommandInterface the possibility to retrieve distance of neighbors

Tambup opened the pull request:
Hi,
another student and me (both of UNIBS) added the possibility to retrieve the distance of neighbors of a vehicle.
This was necessary to verify the possibility to change lane securely, that is checking the distance of other vehicles before beginning the change of the lane of the whole platoon. Furthermore, this allows us to make different test to check which distance can be marked as secure.

:warning: The "CMD_NEIGHBORING_VEHICLES" must be inserted in veins. Its value should be 0xBF.
An alternative could be to hardcode this in plexe, but i think it's not a good idea.

I put a minimal doc, that is similar to the doc in the code of plexe. I guess it would be better to explain in the doc the value that lateralDirection and longitudinalDirection must have (but is already explained in SUMO). We can discuss about it before accepting the pull request.

Commit:
Adding in CommandInterface the possibility to retrieve distance of neighbors.

michele-segata commented:
Hi,
sorry for the very late reply. I would accept the PR but the problem, as far as I understand, is that the code doesn't compile because `CMD_NEIGHBORING_VEHICLES` is not currently in Veins. Am I right? If that is the case I would do something like this:

1. Define the macro with an `#ifndef` in Plexe's `CommandInterface.h` like this:
```
#ifndef CMD_NEIGHBORING_VEHICLES
#define CMD_NEIGHBORING_VEHICLES 0xbf
#endif
```
2. Submit a PR to Veins adding `#define CMD_NEIGHBORING_VEHICLES 0xbf`

This way, future releases of Veins will have the macro defined but Plexe won't break due to a re-definition. In future releases of Plexe we can then remove the definition.

Could you do this?

Tambup commented:
Hi,
PR done, when I'll have a reply from the owner of veins I'll add the macro. 
Anyway, thanks for the attention.
Best.

michele-segata commented:
In the meanwhile, you can add
```
#ifndef CMD_NEIGHBORING_VEHICLES
#define CMD_NEIGHBORING_VEHICLES 0xbf
#endif
```
to `CommandInterface.h` in Plexe and I will merge the PR.

Commit:
Changing name of CMD_NEIGHBORING_VEHICLES to VAR_NEIGHBORS and adding macro if veins used version hasn't this var.

Tambup commented:
Hi,
i changed the name of CMD_NEIGHBORING_VEHICLES to VAR_NEIGHBORS, because the name wasn't conform to the sumo namestyle as sommers pointed out.

I also added the macro in the header file as you suggested.

michele-segata commented:
Merged! Many thanks!
